buildrunner/loggers.py

```diff
@@ -0,0 +1,255 @@
+"""
+Copyright 2024 Adobe
+All Rights Reserved.
+
+NOTICE: Adobe permits you to use, modify, and distribute this file in accordance
+with the terms of the Adobe license agreement accompanying it.
+"""
+
+import logging
+import os
+import sys
+import queue
+from typing import Optional, Union
+
+import colorlog
+from rich import progress
+
+
+CONSOLE_LOGGER_NAME = "buildrunner"
+
+
+class CustomColoredFormatter(colorlog.ColoredFormatter):
+    def __init__(self, fmt: str, no_color: bool, color: str = "white"):
+        super().__init__(
+            fmt, no_color=no_color, log_colors=self._get_log_level_colors(color)
+        )
+        self.fmt = fmt
+        self.color = color
+
+    def clone(
+        self, color: Optional[str] = None, no_color: Optional[bool] = None
+    ) -> "CustomColoredFormatter":
+        if color is None:
+            color = self.color
+        if no_color is None:
+            no_color = self.no_color
+        return CustomColoredFormatter(self.fmt, no_color, color)
+
+    @staticmethod
+    def _get_log_level_colors(color: str) -> dict:
+        return {
+            "DEBUG": color,
+            "INFO": color,
+            "WARNING": color,
+            "ERROR": color,
+            "CRITICAL": color,
+        }
```

Comment:
I am not sure what the purpose of this method. Doesn't it set the same color for all levels?

Reply:
Yes, the logging library assumes that you want different colored lines for each level, so it accounts for that and makes you set it per level. In our case, we want all logs for a logger with a specific formatter to be the same color, so we just set them all to that one color. It's a workaround for how the library handles it.